File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // Creating a hashmap to assist with checking duplicates
        unordered_map<int, int> duplicateChecker;

        // Traversing through all the numbers
        for (int number : nums) {
            // Attempting to find the position of the current number
            // Wonder if we can use the contain method here as well?
            if (duplicateChecker.contains(number)) {
                // If we have not reached the end, it means it has been seen before
                return true;
            }

            // Marking it as seen
            duplicateChecker[number] = 1;
        }

        // Returning false as all numbers have been examined
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Creating a dict
        seen = {}

        # Traversing the list
        for index, num in enumerate(nums):
            # Getting the complement of our current number
            complement = target - num

            # Checking if we have seen this complement before
            if complement in seen:
                # If so, return the index we're currently on as
                # well as the complement's index
                return [seen[complement], index]
            else:
                # If we haven't seen the complement, simply store the
                # index using the number as a key
                seen[num] = index

        # Returning an empty list just in case
        return []
